pulser/simulation/simulation.py

```diff
@@ -151,10 +151,15 @@ def set_config(self, cfg: SimConfig) -> None:
                 )
             # Probability of phase (Z) flip
             prob = self.config.dephasing_prob
-            self._collapse_ops += [
-                np.sqrt(1.0 - prob) * self.op_matrix["I"],
-                np.sqrt(prob)
-                * (self.op_matrix["sigma_rr"] - self.op_matrix["sigma_gg"]),
+            global_id = qutip.tensor(
+                [self.op_matrix["I"] for _ in range(self._size)]
+            )
+            global_sigmaz = self._build_operator(
+                "sigma_rr", global_op=True
+            ) - self._build_operator("sigma_gg", global_op=True)
+            self._collapse_ops = [
+                np.sqrt(1.0 - prob) * global_id,
+                np.sqrt(prob) * global_sigmaz,
```

Comment:
So this was wrong before?

Reply:
Yes, when dealing with multiple qubits the collapse operators have to have the correct shape... We overlooked this case before, dealing with dephasing only on single-qubit systems.